etc/migration/migrateUsersToShibboleth.js

```diff
@@ -0,0 +1,352 @@
+                } else {
+                    util.format('%s users were migrated to Shibboleth logins.', mappedUsers.length);
+                }
+
+                if (errors > 0) {
+                    log().warn('Some users could not be mapped to Shibboleth logins, check the CSV file for more information');
+                }
+
+                return _exit(0);
+            });
+        });
+    });
+});
+
+/**
+ * Page through all the users in the system, and filter them by tenant
+ *
+ * @param  {String}     tenantAlias             The tenant alias to filter by
+ * @param  {Function}   callback                Invoked when users have been collected
+ * @param  {Object[]}   callback.userHashes     An array of user hashes
+ * @api private
+ */
+var _getAllUsersForTenant = function(tenantAlias, callback) {
+    var userHashes = [];
+
+    PrincipalsDAO.iterateAll(['principalId', 'tenantAlias', 'email'], 100, _aggregateUsers, function(err) {
+        if (err) {
+            log().error({'err': err}, 'Failed to iterate all users');
+            return _exit(1);
+        }
+
+        log().info('Found %s users for specified tenant', userHashes.length);
+
+        return callback(userHashes);
+    });
+
+    /*!
+     * Filter users down to those that are part of the specified tenant. Then add them to the `userHashes` array
+     *
+     * @param  {Object[]}   principalHashes     The principals to filter and aggregate
+     * @param  {Function}   callback            Will be invoked when the principals are aggregated
+     */
+    function _aggregateUsers(principalHashes, callback) {
+        log().info('Checking %s principals for tenancy', principalHashes.length);
+        _.chain(principalHashes)
+            .filter(function(principalHash) {
+                return principalHash.tenantAlias === tenantAlias && PrincipalsDAO.isUser(principalHash.principalId) && principalHash.email;
+            })
+            .each(function(userHash) {
+                log().info('Adding user %s to tenant users', userHash.email);
+                userHashes.push(userHash);
+            });
+
+        return callback();
+    }
+};
+
+/**
+ * Get the external login ids for all users in a tenancy
+ *
+ * @param  {Object[]}   userHashes            The users for a tenancy
+ * @param  {Function}   callback              Standard callback function
+ * @param  {Object[]}   callback.userHashes   The users for a tenancy with ID
+ * @api private
+ */
+var _getExternalIds = function(userHashes, callback, _userHashes) {
+    _userHashes = _userHashes || [];
+
+    if (_.isEmpty(userHashes)) {
+        return callback(_userHashes);
+    }
+
+    // Take the next principal hash from the collection
+    var principalHash = userHashes.shift();
+    _findExternalId(principalHash, function(err, userHash) {
+        if (!err && userHash) {
+            // Accumulate the return information we want
+            _userHashes.push(userHash);
+        }
```

Comment:
Basically, an error when trying to retrieve data from Cassandra. We're logging it in `_findExternalId` method and  adding the user to the CSV, and I thought since it'd mostly likely be data error we didn't need to terminate.